esp-hal/src/interrupt/riscv.rs

```diff
@@ -735,40 +727,41 @@ mod plic {
     /// Get interrupt priority - called by assembly code
     #[inline]
     pub(super) unsafe extern "C" fn get_priority(cpu_interrupt: CpuInterrupt) -> Priority {
-        let plic_mxint_pri_ptr = PLIC_MXINT0_PRI_REG as *mut u32;
-
-        let cpu_interrupt_number = cpu_interrupt as isize;
-        let prio = plic_mxint_pri_ptr
-            .offset(cpu_interrupt_number)
-            .read_volatile();
-        core::mem::transmute::<u8, Priority>(prio as u8)
+        let plic = &*crate::peripherals::PLIC_MX::PTR;
+        let prio = plic
+            .mxint_pri(cpu_interrupt as usize)
+            .read()
+            .cpu_mxint_pri()
+            .bits();
+        core::mem::transmute::<u8, Priority>(prio)
     }
     #[no_mangle]
     #[link_section = ".trap"]
     pub(super) unsafe extern "C" fn _handle_priority() -> u32 {
         use super::mcause;
-        let plic_mxint_pri_ptr = PLIC_MXINT0_PRI_REG as *mut u32;
-        let interrupt_id: isize = unwrap!(mcause::read().code().try_into()); // MSB is whether its exception or interrupt.
-        let interrupt_priority = plic_mxint_pri_ptr.offset(interrupt_id).read_volatile();
+        let plic = &*crate::peripherals::PLIC_MX::PTR;
 
-        let thresh_reg = PLIC_MXINT_THRESH_REG as *mut u32;
-        let prev_interrupt_priority = thresh_reg.read_volatile() & 0x000000FF;
-        // this is a u8 according to esp-idf, so mask everything else.
+        let interrupt_id: usize = mcause::read().code(); // MSB is whether its exception or interrupt.
```

Comment:
Just to double-check, the change from `isize` to `usize` is intended?

Reply:
Yes, otherwise it causes errors. `mcause::read().code()` returns `usize`, and then `usize` is expected when accessing the array `plic.mxint_pri(x)`